src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -0,0 +1,260 @@
+import AbstractAlgebra: FieldElem, map, Ring
+import Hecke: evaluate, multiplicative_jordan_decomposition, PolyElem, _rational_canonical_form_setup, refine_for_jordan
+
+export
+    block_matrix,
+    complement,
+    conjugate_transpose,
+    diagonal_join,
+    insert_block,
+    insert_block!,
+    isconjugate_gl,
+    ishermitian_matrix,
+    isskewsymmetric_matrix,
+    permutation_matrix,
+    submatrix
+
+
+
+########################################################################
+#
+# Matrix manipulation
+#
+########################################################################
+
+"""
+    submatrix(A::MatElem{T}, i::Int, j::Int, m::Int, n::Int)
+Return the `m x n` submatrix of `A` rooted at `(i,j)`
```

Comment:
I think there needs to be an empty line after the signature and before the explanation text.
```suggestion

Return the `m x n` submatrix of `A` rooted at `(i,j)`
```